business_objects/monitor.py

```diff
@@ -0,0 +1,424 @@
+) -> None:
+    query = f"""
+    UPDATE embedding
+    SET state = '{enums.EmbeddingState.FAILED.value}'
+    WHERE state = '{enums.EmbeddingState.ENCODING.value}' OR state = '{enums.EmbeddingState.WAITING.value}' OR state = '{enums.EmbeddingState.INITIALIZING.value}'
+    """
+
+    if project_id:
+        query = (
+            query
+            + f"""
+        AND project_id = '{project_id}'
+        """
+        )
+    if embedding_id:
+        query = (
+            query
+            + f"""
+        AND id = '{embedding_id}'
+        """
+        )
+    general.execute(query)
+    general.flush_or_commit(with_commit)
+
+
+def set_weak_supervision_to_failed(
+    project_id: str = None, task_id: str = None, with_commit: bool = False
+) -> None:
+    query = f"""
+    UPDATE weak_supervision_task
+    SET state = '{enums.PayloadState.FAILED.value}'
+    WHERE state = '{enums.PayloadState.STARTED.value}'
+    """
+
+    if project_id:
+        query = (
+            query
+            + f"""
+        AND project_id = '{project_id}'
+        """
+        )
+    if task_id:
+        query = (
+            query
+            + f"""
+        AND id = '{task_id}'
+        """
+        )
+    general.execute(query)
+    general.flush_or_commit(with_commit)
+
+
+def set_upload_task_to_failed(
+    project_id: str = None, task_id: str = None, with_commit: bool = False
+) -> None:
+    query = f"""
+    UPDATE upload_task
+    SET state = '{enums.UploadStates.ERROR.value}'
+    WHERE state = '{enums.UploadStates.IN_PROGRESS.value}' OR state = '{enums.UploadStates.PENDING.value}' OR state = '{enums.UploadStates.PREPARED.value}' OR state = '{enums.UploadStates.WAITING.value}'
+    """
+
+    if project_id:
+        query = (
+            query
+            + f"""
+        AND project_id = '{project_id}'
+        """
+        )
+    if task_id:
+        query = (
+            query
+            + f"""
+        AND id = '{task_id}'
+        """
+        )
+    general.execute(query)
+    general.flush_or_commit(with_commit)
+
+
+def __select_running_information_source_payloads(
+    project_id: str = None, only_running: bool = False
+) -> str:
+    query = f"""
+    SELECT id, 'information_source' task_type, state, project_id, created_by
+    FROM information_source_payload
+    """
+    if project_id and only_running:
+        query = (
+            query
+            + f"""
+        WHERE project_id = '{project_id}' AND state = '{enums.PayloadState.CREATED.value}'
+        """
+        )
+    if project_id:
+        query = (
+            query
+            + f"""
+        WHERE project_id = '{project_id}'
+        """
+        )
+    if only_running:
+        query = (
+            query
+            + f"""
+        WHERE state = '{enums.PayloadState.CREATED.value}'
+        """
+        )
+    query = (
+        query
+        + f"""
+    LIMIT 100
+    """
+    )
+    return query
```

Comment:
since this will be harder and harder to extend id go with extension logic proposition (untested though :D ):
```suggestion
    where_add = ""
    if project_id:
        where_add = __extend_where(where_add, f"project_id = '{project_id}'")
    if only_running:
        where_add = __extend_where(
            where_add, f"state = '{enums.PayloadState.CREATED.value}'"
        )
    query += where_add
    query += "\nLIMIT 100"
    return query
        
        
 #function outside code block
 def __extend_where(current:str,condition:str) -> str:
    if current:
        return f"{current} AND {condition}"
    else:
        return f" WHERE {condition}"
```

Reply:
similar for the others on the page

Reply:
I understand, this is a good solution, but maybe we wan't to tackle this if we are going to extend this? Because if we ain't gonna need this I would rather keep it simple

Reply:
to be honest I found the current solution irritating and it took me a bit to understand (because of multiple conditions for the same parameter & no if/elif/else logic). Also, I don't think the current solution would work since for both project_id and only_running set this would result in multiple WHERE parts. 

The proposed solution will reduce the lines of code in the whole file, increase readability and less error-prone so I'd actually say it is the simpler one (though ofc this can be a personal impression)

Reply:
Build in your solution a litte bit adapted